service/exporter/exporter.go

```diff
@@ -57,12 +20,52 @@ type namespace struct {
 	Name string
 	Pods []pod
 }
-type leanixExport struct {
+type Response struct {
 	Namespaces []namespace
 	LastUpdate time.Time
 }
 
-func getNamespaces(c *kubernetes.Clientset) []namespace {
+type Config struct {
+	Excludes []string
+}
+
+// New creates a new configured version service.
+func New(config Config) (*Service, error) {
+	// Settings.
+	if config.Excludes == nil {
+		return nil, microerror.MaskAnyf(invalidConfigError, "config.Excludes must not be empty")
+	}
+	newService := &Service{
+		Config: config,
+	}
+
+	return newService, nil
+}
+
+// Service implements the version service interface.
+type Service struct {
+	Config
+}
+
+func (s *Service) Get(ctx context.Context) (*Response, error) {
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
```

Comment:
In this case the latter one. In case you have to return your own error it should be a typed one where we also have a matcher function for. This is not necessary here because you receive any. What you could do is annotating the error messages with some custom information but this is also not necessary here.